src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -183,8 +188,28 @@ internal override RefKind RefKind
         {
             get
             {
-                ComputeReturnType();
-                return _lazyRefKind;
+                ComputeRefKind();
+                return _lazyRefKind.Value;
+            }
+        }
+
+        internal void ComputeRefKind()
+        {
+            if (_lazyRefKind.HasValue)
+            {
+                return;
+            }
+
+            if (_syntax.ReturnType.Kind() == SyntaxKind.RefType)
```

Comment:
>if (_syntax.ReturnType.Kind() == SyntaxKind.RefType) [](start = 12, length = 52)

It looks like all calculation here is syntax based, I do not believe it makes sense to do this lazily, we can reliably calculate this in constructor.